src/mbgl/gl/enum.cpp

```diff
@@ -0,0 +1,225 @@
+#include <mbgl/gl/enum.hpp>
+#include <mbgl/gfx/types.hpp>
+#include <mbgl/gl/defines.hpp>
+
+namespace mbgl {
+namespace gl {
+
+template <>
+platform::GLenum Enum<gfx::PrimitiveType>::to(const gfx::PrimitiveType value) {
+    switch (value) {
+        case gfx::PrimitiveType::Points: return GL_POINTS;
+        case gfx::PrimitiveType::Lines: return GL_LINES;
+        case gfx::PrimitiveType::LineLoop: return GL_LINE_LOOP;
+        case gfx::PrimitiveType::LineStrip: return GL_LINE_STRIP;
+        case gfx::PrimitiveType::Triangles: return GL_TRIANGLES;
+        case gfx::PrimitiveType::TriangleStrip: return GL_TRIANGLE_STRIP;
+        case gfx::PrimitiveType::TriangleFan: return GL_TRIANGLE_FAN;
+    }
+    return GL_INVALID_ENUM;
+}
+
+template <>
+gfx::ColorBlendEquationType Enum<gfx::ColorBlendEquationType>::from(const platform::GLint value) {
+    switch (value) {
+        case GL_FUNC_ADD: return gfx::ColorBlendEquationType::Add;
+        case GL_FUNC_SUBTRACT: return gfx::ColorBlendEquationType::Subtract;
+        case GL_FUNC_REVERSE_SUBTRACT: return gfx::ColorBlendEquationType::ReverseSubtract;
+    }
+    return {};
```

Comment:
should this be handled somehow? assert or log error?

Reply:
I don't think so; in that case it defaults to an undefined value of `gfx::ColorBlendEquationType` which will be okay for most use cases. Additionally, we're only using the OpenGL state getters for debug purposes; they're not used in production builds.